omod-1.8/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/resource/openmrs1_8/ModuleActionResource1_8.java

```diff
@@ -0,0 +1,225 @@
+import org.openmrs.module.webservices.rest.web.annotation.PropertyGetter;
+import org.openmrs.module.webservices.rest.web.annotation.Resource;
+import org.openmrs.module.webservices.rest.web.representation.Representation;
+import org.openmrs.module.webservices.rest.web.resource.api.Creatable;
+import org.openmrs.module.webservices.rest.web.resource.impl.BaseDelegatingResource;
+import org.openmrs.module.webservices.rest.web.resource.impl.DelegatingResourceDescription;
+import org.openmrs.module.webservices.rest.web.response.IllegalRequestException;
+import org.openmrs.module.webservices.rest.web.response.ResourceDoesNotSupportOperationException;
+import org.openmrs.module.webservices.rest.web.response.ResponseException;
+
+import javax.servlet.ServletContext;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+
+@Resource(name = RestConstants.VERSION_1 + "/moduleaction", supportedClass = ModuleAction.class, supportedOpenmrsVersions = {
+        "1.8.*", "1.9.*", "1.10.*", "1.11.*", "1.12.*", "2.0.*", "2.1.*" })
+public class ModuleActionResource1_8 extends BaseDelegatingResource<ModuleAction> implements Creatable {
+	
+	/**
+	 * ModuleFactoryWrapper is used for testing purposes.
+	 */
+	private ModuleFactoryWrapper moduleFactoryWrapper = new ModuleFactoryWrapper();
+	
+	public void setModuleFactoryWrapper(ModuleFactoryWrapper moduleFactoryWrapper) {
+		this.moduleFactoryWrapper = moduleFactoryWrapper;
+	}
+	
+	/**
+	 * Overriding create directly, because ModuleFactory requires ServletContext to execute any
+	 * action
+	 */
+	@Override
+	public Object create(SimpleObject post, RequestContext context) throws ResponseException {
+		moduleFactoryWrapper.checkPrivilege();
+		
+		ModuleAction action = newDelegate();
+		setConvertedProperties(action, post, getCreatableProperties(), true);
+		
+		Collection<Module> modules;
+		if (action.isAllModules() != null && action.isAllModules()) {
+			modules = moduleFactoryWrapper.getLoadedModules();
+			action.setModules(new ArrayList<Module>(modules));
+		} else {
+			modules = action.getModules();
+		}
+		
+		ServletContext servletContext = getServletContext(context);
+		
+		if (modules == null || modules.isEmpty()) {
+			throw new IllegalRequestException("Cannot execute action " + action.getAction() + " on empty set of modules.");
+		} else {
+			//rest module cannot be stopped/unloaded during handling request
+			Module restModule = moduleFactoryWrapper.getModuleById(RestConstants.MODULE_ID);
+			modules.remove(restModule);
+			switch (action.getAction()) {
+				case START:
+					startModules(modules, servletContext);
+					break;
+				case STOP:
+					stopModules(modules, servletContext, true);
+					break;
+				case RESTART:
+					restartModules(modules, servletContext);
+					break;
+				case UNLOAD:
+					unloadModules(modules, servletContext);
+					break;
+			}
+		}
+		
+		return ConversionUtil.convertToRepresentation(action, Representation.DEFAULT);
```

Comment:
Client is just sending a couple of uuids and action string, i thought it would be nice to return a bit more info about affected modules, similiary how we return created object in response body when handling POST request.